### PR TITLE
Add wallet public key mapping to session data in IndexedDB for multi-wallet support

### PR DESCRIPTION
**Pull request by vjgpt**

### CODE REVIEW

File: packages/gpl-core/src/idl/gpl_core.json [hunk 1]
```diff
@@ -1756,24 +1756,12 @@
...
-          },
-          {
-            "name": "UserError"
-          },
-          {
-            "name": "ReactionError"
-          },
-          {
-            "name": "ConnectionError"
+            "name": "URITooLong"
```
Comment (abishekk92) [inline packages/gpl-core/src/idl/gpl_core.json:1764]: This is the new IDL?

File: packages/react-sdk/src/utils/indexedDB.ts [hunk 2]
```diff
@@ -19,12 +21,12 @@ export const openIndexedDB = async () => {
...
 };
 
-export const getItemFromIndexedDB = async (storeName: string) => {
+export const getItemFromIndexedDB = async (storeName: string, key?: IDBValidKey) => {
   const db = await openIndexedDB();
   return new Promise<any>((resolve, reject) => {
     const transaction = db.transaction(storeName, 'readonly');
     const store = transaction.objectStore(storeName);
-    const request = store.get(1);
+    const request = key ? store.get(key) : store.get(1);
```
Comment (abishekk92) [inline packages/react-sdk/src/utils/indexedDB.ts:29]: What does this fallback logic do?

File: packages/gpl-core/src/idl/gpl_core.json (hunk 1 above)
Comment (vjgpt) [reply packages/gpl-core/src/idl/gpl_core.json:1764]: yes this is the new IDL

File: packages/react-sdk/src/utils/indexedDB.ts (hunk 2 above)
Comment (vjgpt) [reply packages/react-sdk/src/utils/indexedDB.ts]: pushed new changes that enforce the key param in the function so we don't need the fallback logic now.

**Review by abishekk92 — APPROVED** [review]

**Pull request merged**